django/db/models/expressions.py

```diff
@@ -902,7 +902,21 @@ def resolve_expression(
         return query.resolve_ref(self.name, allow_joins, reuse, summarize)
 
     def replace_expressions(self, replacements):
-        return replacements.get(self, self)
+        if (replacement := replacements.get(self)) is not None:
+            return replacement
+        field_name, *transforms = self.name.split(LOOKUP_SEP)
+        if not transforms:
+            return self
```

Comment:
I wonder if this is needed :thinking: It's not covered by tests and, as far as I'm aware, it's already handled by `if (replacement := replacements.get(F(field_name))) is None ...`.